src/language-markdown/parser-markdown.js

```diff
@@ -59,7 +63,23 @@ function htmlToJsx() {
         return node;
       }
 
-      return Object.assign({}, node, { type: "jsx" });
+      const htmlNodes = parseHtml(node.value).children.filter(
+        node => node.type === "element"
+      );
+
+      // find out if there are adjacent JSX elements which should be allowed in mdx alike in markdown
+      if (htmlNodes.length <= 1) {
+        return Object.assign({}, node, { type: "jsx" });
+      }
+
+      return htmlNodes.map(htmlNode => ({
+        type: "jsx",
+        value: printDocToString(
```

Comment:
Anyone knows how to get the original text from ast `htmlNode`? Or is a better way to split the elements text?

Reply:
OK, I finally find out that it's so easy to get the original text from ast position. 😂